packages/strapi-admin/admin/src/utils/FieldApi.js

```diff
@@ -0,0 +1,36 @@
+import { cloneDeep } from 'lodash';
+import invariant from 'invariant';
+
+class FieldApi {
+  fields = {};
+
+  getField = type => {
+    invariant(type, 'A type must be provided');
+
+    return cloneDeep(this.fields[type]) || null;
+  };
+
+  getFields = () => {
+    return cloneDeep(this.fields);
+  };
+
+  registerField = field => {
```

Comment:
As this function is used to change the FieldApi property, you can create a `setter`.